lib/matplotlib/colorbar.py

```diff
@@ -802,8 +802,9 @@ def _process_values(self, b=None):
 
             b = self.norm.inverse(self._uniform_y(self.cmap.N + 1))
 
-            if isinstance(self.norm, colors.LogNorm):
-                # If using a lognorm, ensure extensions don't go negative
+            if isinstance(self.norm, (colors.PowerNorm, colors.LogNorm)):
```

Comment:
I feel like this kind of stuff should be handled by the specific `Norm` class, instead of being "special cased" here in this code, but since this is how `LogNorm` is dealt with at the moment I guess this is fine as a fix.

Reply:
Hmmm, interesting idea.  This "b" is just the points that are used to make the pcolor mesh that makes the colorbar.  So its a bit of a toss up where making that list should get created and how.